state/policy.go

```diff
@@ -0,0 +1,58 @@
+package state
+
+import (
+	"github.com/hashicorp/nomad-autoscaler/state/policy"
+)
+
+// policyUpdateHandler handles all updates from the policy.Watcher and is
+// responsible for all safety checks and sanitization before the policy is
+// placed into our internal state.
+func (h *Handler) policyUpdateHandler() {
+	for {
+		select {
+		case p := <-h.policyUpdateChan:
+
+			jobID := p.Target["Job"]
+
+			// Ensure the scale status watcher is running for the job.
+			h.startStatusWatcher(jobID)
+
+			// If the job is stopped, we don't need to work on storing the policy
+			// and should exit.
+			if h.statusState.IsJobStopped(jobID) {
+				h.log.Debug("job in stopped state, skipping policy update", "job_id", jobID)
+				break
+			}
+
+			// TODO(jrasell) once we have a better method for surfacing errors to the
+			//  user, this error should be presented.
+			if err := policy.Validate(p); err != nil {
+				h.log.Error("failed to validate policy", "error", err, "policy_id", p.ID)
+				break
+			}
+
+			if p.Policy[policy.KeySource] == nil {
+				p.Policy[policy.KeySource] = ""
+			}
+
+			autoPolicy := &policy.Policy{
```

Comment:
It took me a bit to realize that `p` is a Nomad API policy object and that we are transforming it into `autoPolicy`, which is the Autoscaler internal representation of a policy.

Maybe have a short comment before explaining this.

Reply:
added a comment to my local changes.